### PR TITLE
Search in descriptions of courses if applicable (fixed formatting)

### PR DESCRIPTION
**Pull request by GoldinGuy**

### CODE REVIEW

File: frontend/src/contexts/searchContext.tsx [hunk 1]
```diff
@@ -714,22 +714,19 @@ export const SearchProvider: React.FC = ({ children }) => {
...
-            professor.toLowerCase().includes(token)
-          )
-        )
-          continue;
+        if (listing.subject.toLowerCase().startsWith(token) ||
+	listing.number.toLowerCase().startsWith(token) ||
+	// for course numbers that start with a letter
+	// (checked by if .toLowerCase() is not equal to .toUpperCase(), see https://stackoverflow.com/a/32567789/5540324),
+	// exclude this letter when comparing with the search token
+	(numberFirstChar.toLowerCase() !== numberFirstChar.toUpperCase() &&
```
[the remark this answers is not in the document]
Comment (GoldinGuy) [reply frontend/src/contexts/searchContext.tsx:722]: What is this referring to?

[the remark this answers is not in the document]
Comment (Josh-Cena) [reply frontend/src/contexts/searchContext.tsx:722]: The hack of `numberFirstChar.toLowerCase() !== numberFirstChar.toUpperCase()`. This doesn't work for anything other than ASCII, which is the case here, but there's a way that's both more concise and more robust.

[the remark this answers is not in the document]
Comment (GoldinGuy) [reply frontend/src/contexts/searchContext.tsx:722]: Ah. That is not a line of code I contributed, is there a reason it's holding up this PR?

**Review by Josh-Cena — APPROVED** [review]
I see—you changed the formatting so it wasn't as obvious. Re-ran formatting and now I see the change. This LGTM

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
Actually, it cannot be shipped like this. Descriptions can contain arbitrary text and it will be very confusing when a lot more results show up while they do not look immediately related (because descriptions are only visible after you click on the entry). We need to hide this behind a checkbox.

**Comment by Josh-Cena** [comment]
Hi @GoldinGuy thanks for the PR. There is https://github.com/coursetable/coursetable/pull/1353 also, and it seems much closer to the finish line (that one is implemented behind an option). As such I'll close this, but still thanks for the effort.

**Pull request closed without merging**